README.md

```diff
@@ -1,116 +1,170 @@
 
-# Training loop
-for epoch in range(num_epochs):
-    # Training process
-    ...
-    scheduler.step()  # Update learning rate
-```
+*Note:* In most cases, YAML does not expect strings. Adding quotation marks around arguments in the config file can lead to unexpected errors.
 
-#### 3. **Random Search for Hyperparameter Tuning**:
+### 3: Script Execution
 
-Pseudo-code for implementing random search.
+Once a YAML config file is prepared, it can be passed into any script in the pipeline. Before you run any scripts, it is recommended to copy all of them into the [user_scripts](./scripts/user_scripts/) directory and modify the scripts to point to the right config file.
 
-```python
-import random
+The expected order of script execution is as follows:
+1. Change current directory to `scripts/user_scripts`.
+2. Run `download_data.sh` to fetch the necessary data.
+3. Execute `train_tokenizer.sh` to prepare the tokenizer.
+4. Use `tokenize_data.sh` for data tokenization.
+5. Finally, run `retnet.sh` or `transformer.sh`, depending on your project needs.
 
-def random_search_hyperparams():
-    learning_rates = [1e-3, 1e-4, 1e-5]
-    batch_sizes = [32, 64, 128]
-    num_layers_options = [2, 4, 6]
+For example, if you want to train a RetNet model:
 
-    # Randomly select hyperparameters
-    lr = random.choice(learning_rates)
-    batch_size = random.choice(batch_sizes)
-    num_layers = random.choice(num_layers_options)
+```bash
+cd scripts/user_scripts
 
-    # Create and return a model with these hyperparameters
-    model = create_model(lr, batch_size, num_layers)
-    return model
+./download_data.sh
+./train_tokenizer.sh
+./tokenize_data.sh
 
-# Example usage
-tuned_model = random_search_hyperparams()
-# Train and evaluate this model
+# Replace this with the file pertaining to the model that you want to run
+./retnet.sh
 ```
 
-### Command Line Examples
+## Features
 
-#### 1. **Running a Training Script with Hyperparameters**:
+### Grid Search
 
-Assuming you have a training script `train.py`, you can run it with different hyperparameters.
+The Grid Search feature is designed to systematically explore a range of hyperparameters and compare RetNet and Transformer models with corresponding parameters at each point. This evaluates both architectures with various combinations of learning rates, embedding dimensions, feed-forward dimensions, sequence lengths, and number of heads. The goal is to identify the configuration that results in the best model performance, measured in terms of loss and training efficiency.
+
+**Code Overview:**
+
+We implement the grid search process as follows:
+
+- **Hyperparameters Tested:** Learning rates (`0.001`, `0.0005`, `0.0001`), embedding dimensions (`768`, `1024`, `1280`), feed-forward dimensions (`1024`, `2048`), heads (`4`, `8`), and sequence lengths (`256`, `512`) for a total of 72 unique combinations.
```

Comment:
for a total of 72 unique combinations
--> change to 
for a total of 72 unique combinations per model architecture.

Reply:
Others commenting would be nice on the matter.

Reply:
I will change that.